mesonbuild/interpreter.py

```diff
@@ -3631,6 +3631,25 @@ def run(self):
         mlog.log('Build targets in project:', mlog.bold(str(len(self.build.targets))))
         FeatureNew.called_features_report()
         FeatureDeprecated.called_features_report()
+        if self.subproject == '':
+            self.print_extra_warnings()
+
+    def print_extra_warnings(self):
+        for c in self.build.compilers.values():
+            if c.get_id() == 'clang':
+                self.check_clang_asan_lundef()
+                break
+
+    def check_clang_asan_lundef(self):
+        if 'b_lundef' not in self.coredata.base_options:
+            return
+        if 'b_sanitize' not in self.coredata.base_options:
+            return
+        if 'address' in self.coredata.base_options['b_sanitize'].value:
+            if self.coredata.base_options['b_lundef'].value:
+                mlog.warning('''Trying to use address sanitizer on Clang with b_lundef.
+This will probably not work.
+Try setting b_lundef to false instead.''')
```

Comment:
Maybe this should be done in `interpreter.py:emit_base_options_warnings()` ?

Reply:
Maybe it should, but those errors are (currently at least) printed when the options are added. It's easy to lose those in the noise. I put the code in its current location so it would be printed at the very end.

Maybe we could add a method like `store_warning_message_to_be_printed_after_successful_run`?

Reply:
Any opinions on this?